tests/test_buffering.py

```diff
@@ -603,6 +603,25 @@ def test_multi_access():
     assert np.all(w.data == w1.data)
 
 
+def test_issue_1901():
+    grid = Grid(shape=(2, 2))
+    time = grid.time_dim
+    x, y = grid.dimensions
+
+    usave = TimeFunction(name='usave', grid=grid, save=10)
+    v = TimeFunction(name='v', grid=grid)
+
+    eq = [Eq(v[time, x, y], usave)]
```

Comment:
nitpicking: `Eq(usave, v)` bit more readable and makes bit more "sense"

Reply:
I want `time` in place of `t` since otherwise the loop will be tagged SEQUENTIAL due to the presence of uindices. But I want it fully PARALLEL to trigger the issue

Reply:
Yeah that's fin,e I meant having `usave` as lhs mostly but again just nitpicking

Reply:
ah, I see what you mean now... 
well in backward propagators you do have usave's on the RHS :)